packages/client/utils/AtlassianManager.ts

```diff
@@ -697,8 +755,28 @@ export default abstract class AtlassianManager {
     tabId: string,
     fieldId: string
   ) {
-    return (await this.delete(
+    const res = (await this.delete(
       `https://api.atlassian.com/ex/jira/${cloudId}/rest/api/3/screens/${screenId}/tabs/${tabId}/fields/${fieldId}`
     )) as null | AtlassianError | JiraError
+
+    if (res === null) {
+      return null
+    }
+
+    if (isAtlassianError(res)) {
+      throw new Error(res.message)
```

Comment:
if API errors get thrown, then you'll have to wrap any method call in a try/catch and you can't use Promise.all because that will throw if any of the promises throw.

to keep things clean, i'd treat an error message as a normal return value. By returning a discriminated union, typescript is really good at determining the specific type by doing something like `if ('errors' in res) ...`

Reply:
@mattkrick, I see. Would you rather see error handling directly in the function that calls these methods, i.e., `addMissingJiraField` in this case? I didn't want to return the error messages from the new `AtlassianManager` methods to keep things consistent. 

Just to clarify, I'd change that to
```
async addFieldToScreenTab(cloudId: string, screenId: string, tabId: string, fieldId: string) {
    return (await this.post(
      `https://api.atlassian.com/ex/jira/${cloudId}/rest/api/3/screens/${screenId}/tabs/${tabId}/fields/`,
      {fieldId}
    )) as JiraAddScreenFieldResponse | AtlassianError | JiraError
```

and then use helpers like `isAtlassianError` in `addMissingJiraField` function. Any thoughts? 

And btw, what do you think about helpers like 
```
function isJiraNoAccessError<T>(response: T | JiraNoAccessError): response is JiraNoAccessError {
  return 'errorMessages' in response
}
```
Should I leave it or remove it and use check like `'errorMessages' in response` directly? Thanks!

Reply:
https://github.com/ParabolInc/parabol/pull/4880/commits/6a841e0c667b10b2ee98293c221113741cf01acc @mattkrick I think what I think works best here I'd appreciate if you could take a look 🙏

Reply:
big picture, if a call to a REST endpoint returns an error, should we throw it or return it?
I like returning errors when possible. That's because typescript can tell you the return value is e.g. `JiraError | JiraResult` but if you throw, then it becomes `never | JiraResult` and the caught error is of type `unknown`. type safety for the win!

> And btw, what do you think about helpers like

i don't have a strong opinion either way. They both offer the same type deduction in 1 line (if you write `if ('errorMessages' in response)` typescript is still going to be able to infer the type because `errorMessages` is a discriminator). The helper is more expressive, but at the expense of a little extra mental overhead because the debugger will have to see what the function does.

since we're likely going to handle AtlassianError and JiraError in the same way, you could format one so it looks like the other (e.g. `if ('errorMessages' in res) return {message: [res.errorMessages[0]}`) and that way the calling code would be less verbose because it'd only have to check for `JiraError | JiraResult` instead of `AtlassianError | JiraError | JiraResult`.

Reply:
Thanks! 

Regarding the error formatting, that'd make code way more readable. As Jira API errors can be under different properties and formatting one as another could be unnecessarily cumbersome I think the best way is to create one more type like
```
/**
 * Jira API throws a lot of different errors that could be placed in a different properties.
 * This type is used to expose the error under a simple interface for others to consume.
 */
export interface JiraApiError {
  errorMessage: string
}
```
This would make our `AtlassianManager` responsible to format error that way. This way handling errors in resolvers is as easy as
```
    const screensResponse = await manager.getScreens(cloudId)
    if (isJiraApiError(screensResponse)) {
      return {error: {message: screensResponse.errorMessage}}
    }
```
See the: https://github.com/ParabolInc/parabol/pull/4880/commits/520318ee9bb02c7483e3358eb839e57094aa7edb